Content.Server/Anomaly/Effects/LiquidAnomalySystem.cs

```diff
@@ -0,0 +1,128 @@
+using Content.Shared.Chemistry.Components;
+using Content.Server.Fluids.EntitySystems;
+using Robust.Shared.Prototypes;
+
+namespace Content.Server.Anomaly.Effects;
+/// <summary>
+/// This component allows the anomaly to chase a random instance of the selected type component within a radius.
+/// </summary>
+public sealed class LiquidAnomalySystem : EntitySystem
+{
+    [Dependency] private readonly SharedAudioSystem _audio = default!;
+    [Dependency] private readonly EntityLookupSystem _lookup = default!;
+    [Dependency] private readonly SolutionContainerSystem _solutionContainerSystem = default!;
+    [Dependency] private readonly PointLightSystem _light = default!;
+    [Dependency] private readonly IRobustRandom _random = default!;
+    [Dependency] private readonly PuddleSystem _puddleSystem = default!;
+    [Dependency] private readonly IPrototypeManager _proto = default!;
+
+    /// <inheritdoc/>
+    public override void Initialize()
+    {
+        SubscribeLocalEvent<LiquidAnomalyComponent, AnomalyPulseEvent>(OnPulse);
+        SubscribeLocalEvent<LiquidAnomalyComponent, AnomalySupercriticalEvent>(OnSupercritical);
+        SubscribeLocalEvent<LiquidAnomalyComponent, AnomalySeverityChangedEvent>(OnSeverityChanged);
+        SubscribeLocalEvent<LiquidAnomalyComponent, MapInitEvent>(OnMapInit);
+    }
+    private void OnMapInit(EntityUid uid, LiquidAnomalyComponent component, MapInitEvent args)
+    {
+        ChangeReagentType(uid, component);
+    }
+
+    private void OnPulse(EntityUid uid, LiquidAnomalyComponent component, ref AnomalyPulseEvent args)
+    {
+        PulseScalableEffect(
+            uid,
+            component.MaxSolutionGeneration * args.Severity,
+            component.MaxSolutionInjection * args.Severity,
+            component.InjectRadius,
+            component.Reagent);
+    }
+
+    private void OnSupercritical(EntityUid uid, LiquidAnomalyComponent component, ref AnomalySupercriticalEvent args)
+    {
+        PulseScalableEffect(
+            uid,
+            component.SuperCriticalSolutionGeneration,
+            component.SuperCriticalSolutionInjection,
+            component.SuperCriticalInjectRadius,
+            component.Reagent);
+    }
+
+
+    private void OnSeverityChanged(EntityUid uid, LiquidAnomalyComponent component, ref AnomalySeverityChangedEvent args)
+    {
+        //If after the severity change, its level has exceeded the threshold, the type of reagent changes, and the threshold increases.
+        if (args.Severity >= component.NextChangeThreshold)
+        {
+            component.NextChangeThreshold += component.ReagentChangeStep;
+
+            ChangeReagentType(uid, component);
+            return;
+        }
+    }
+
+    private void ChangeReagentType(EntityUid uid, LiquidAnomalyComponent component)
+    {
+        if (component.PossibleChemicals.Count == 0) return;
+
+        var rndIndex = _random.Next(0, component.PossibleChemicals.Count - 1);
```

Comment:
replace this with blacklist code, keep picking a random reagent until it isnt blacklisted (or first remove blacklisted reagents from the list of all of them then pick from that after)

Reply:
![image](https://github.com/space-wizards/space-station-14/assets/96445749/81b1e29a-48a2-432c-94aa-740df8ea1a1a)
@deltanedas 
How to transform the data in this case?

Reply:
chem.ID

also i still dont think it should store on the component, do something like

```
var chemicals = _proto.EnumeratePrototypes<ReagentPrototype>().ToHashSet();
foreach (var chem in component.BlacklistChemicals)
{
    chemicals.Remove(chem);
}
```

then pick from `chemicals`

Reply:
@deltanedas 
But this also means that you will have to calculate the list of available reagents every time the anomaly changes. Isn't it better to write it down 1 time, given that it will be 1 (well, 2 maximum) anomalies on the map?

Reply:
sounds ok to do it on the comp then i guess

Reply:
In the end, I did it your way
¯\_(ツ)_/¯